src/ClientAuthentication.ts

```diff
@@ -69,6 +69,8 @@ export default class ClientAuthentication {
     sessionId: string,
     options: ILoginInputOptions
   ): Promise<void> => {
+    // In order to get a clean start, make sure that the session is logged out on login.
+    await this.logout(sessionId);
```

Comment:
So you are calling `logout()` from within the `login()` function? I thought this would just call `clearSession()`...?

Reply:
Currently logout is basically just calling `clearSession` , but this might change depending of how we store session-related information in the future, and how the underlying libraries that we use implement logout. If `login` implements the session-clearing logic, there will be some redundancy with `logout`.

Reply:
And actually, perhaps this 'clearing out the current session' logic (which you're doing now by calling a 'logout' function) actually belongs inside the 'login handler' itself, and not here at this 'ClientAuthentication' level...? i.e. maybe I have an implementation of a login handler that wants access to the current session info (if any) before it goes ahead and actually performs it's login voodoo...?

Reply:
| Currently logout is basically just calling clearSession 
Ok, but that's not really relevant here I don't think. What is the 'logic' we want here for a login() operation - i.e. is it that we want to perform a formal logout operation first, or is it that we want to perform that login with fresh 'state'? i.e. would our documentation state that login actually performs a logout operation first?
|  there will be some redundancy with logout.
But that's why you factored out the 'clear session' logic into it's own separate function, right (so they're be no redundancy)?

Reply:
It occurred to me that logout _also_ emits an event, so maybe it makes more sense to just clear the session.

Reply:
This is implemented in a27e30bb1563bab3f5fc78e0fcfc0c7e21d2fbb8. It has a couple of side-effects, but I find it to make the code cleaner: the `LogoutHandler` no longer depends on the local storage, but rather on the `SessionInfoManager`, where the clean up logic is implemented. And initiating login also just calls the same `SessionInfoManager::clear` function.